app/controllers/work_move_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# a controller for changing the collection of work.
+class WorkMoveController < ApplicationController
+  before_action :authenticate_user!
+  verify_authorized except: %i[search]
+
+  def search
+    druid = params[:druid]
+    druid = "druid:#{druid}" unless druid.start_with?('druid:')
+    work = Work.find(params[:id])
+    collections = Collection.where(druid:)
+
+    render json: collections.map { |collection| collection_json(collection, work) }
+  end
+
+  def edit
+    authorize! :work_owner
+  end
+
+  def update
+    authorize! :work_owner
```

Comment:
I believe this should be `authorize! work, to: :move_collection?`  you could possibly add:
`authorize! work, to: :move_collection?, context: {collection: collection}`